src/App.tsx

```diff
@@ -1,95 +1,99 @@
-) {
-  // To avoid the original array mutation
-  const visibleGoods = [...goods];
+  sortByLength = () => {
+    this.setState({ sortType: SortType.LENGTH });
+  };
 
-  // Sort and reverse goods if needed
-  // eslint-disable-next-line no-console
-  console.log(sortType, isReversed);
+  resetList = () => {
+    this.setState({
+      isReversed: false,
+      sortType: SortType.NONE,
+    });
+  };
 
-  return visibleGoods;
-}
+  render(): React.ReactNode {
+    const { isReversed, sortType } = this.state;
 
-// DON'T save goods to the state
-// type State = {
-//   isReversed: boolean,
-//   sortType: SortType,
-// };
+    const goods = getReorderedGoods(goodsFromServer, { sortType, isReversed });
 
-export const App: React.FC = () => {
-  return (
-    <div className="section content">
-      <div className="buttons">
-        <button
-          type="button"
-          className="button is-info is-light"
-        >
-          Sort alphabetically
-        </button>
+    return (
+      <div className="section content">
+        <div className="buttons">
+          <button
+            type="button"
+            onClick={this.sortByAlphabet}
+            className={cn('button is-info',
+              { 'is-light': sortType !== SortType.ALPHABET })}
+          >
+            Sort alphabetically
+          </button>
 
-        <button
-          type="button"
-          className="button is-success is-light"
-        >
-          Sort by length
-        </button>
+          <button
+            type="button"
+            onClick={this.sortByLength}
+            className={cn('button is-success',
+              { 'is-light': sortType !== SortType.LENGTH })}
+          >
+            Sort by length
+          </button>
 
-        <button
-          type="button"
-          className="button is-warning is-light"
-        >
-          Reverse
-        </button>
+          <button
+            type="button"
+            onClick={this.reverse}
+            className={cn('button is-warning',
+              { 'is-light': !isReversed })}
+          >
+            Reverse
+          </button>
 
-        <button
-          type="button"
-          className="button is-danger is-light"
-        >
-          Reset
-        </button>
-      </div>
+          {(isReversed
+          || sortType !== SortType.NONE)
+           && (
+             <button
+               type="button"
+               onClick={this.resetList}
+               className="button is-danger is-light"
+             >
+               Reset
+             </button>
+           )}
+
+        </div>
 
-      <ul>
         <ul>
-          <li data-cy="Good">Dumplings</li>
-          <li data-cy="Good">Carrot</li>
-          <li data-cy="Good">Eggs</li>
-          <li data-cy="Good">Ice cream</li>
-          <li data-cy="Good">Apple</li>
-          <li data-cy="Good">...</li>
+          {goods.map(good => (
+            <li
+              data-cy="Good"
+              key={good}
+            >
```

Comment:
no need to move attributes to the new line here, you have only two of them and they will perfectly fit into a single line